content-src/asrouter/templates/FirstRun/FirstRun.jsx

```diff
@@ -0,0 +1,227 @@
+  "browser/branding/sync-brand.ftl",
+  "browser/newtab/onboarding.ftl",
+];
+
+export const helpers = {
+  selectInterruptAndTriplets(message = {}) {
+    const hasInterrupt = Boolean(message.content);
+    const hasTriplets = Boolean(message.bundle && message.bundle.length);
+    const UTMTerm = message.utm_term || "";
+    return {
+      hasTriplets,
+      hasInterrupt,
+      interrupt: hasInterrupt ? message : null,
+      triplets: hasTriplets ? message.bundle : null,
+      UTMTerm,
+    };
+  },
+
+  addFluent(document) {
+    FLUENT_FILES.forEach(file => {
+      const link = document.head.appendChild(document.createElement("link"));
+      link.href = file;
+      link.rel = "localization";
+    });
+  },
+
+  async fetchFlowParams({ fxaEndpoint, UTMTerm, dispatch, setFlowParams }) {
+    try {
+      const url = new URL(
+        `${fxaEndpoint}/metrics-flow?entrypoint=activity-stream-firstrun&form_type=email`
+      );
+      addUtmParams(url, UTMTerm);
+      const response = await fetch(url, { credentials: "omit" });
+      if (response.status === 200) {
+        const { deviceId, flowId, flowBeginTime } = await response.json();
+        setFlowParams({ deviceId, flowId, flowBeginTime });
+      } else {
+        dispatch(
+          ac.OnlyToMain({
+            type: at.TELEMETRY_UNDESIRED_EVENT,
+            data: {
+              event: "FXA_METRICS_FETCH_ERROR",
+              value: response.status,
+            },
+          })
+        );
+      }
+    } catch (error) {
+      dispatch(
+        ac.OnlyToMain({
+          type: at.TELEMETRY_UNDESIRED_EVENT,
+          data: { event: "FXA_METRICS_ERROR" },
+        })
+      );
+    }
+  },
+};
+
+export class FirstRun extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.didLoadFlowParams = false;
+
+    this.state = {
+      prevMessage: undefined,
+
+      hasInterrupt: false,
+      hasTriplets: false,
+
+      interrupt: undefined,
+      triplets: undefined,
+
+      isInterruptVisible: false,
+      isTripletsContainerVisible: false,
+      isTripletsContentVisible: false,
+
+      UTMTerm: "",
+
+      flowParams: undefined,
+    };
+
+    this.closeInterrupt = this.closeInterrupt.bind(this);
+    this.closeTriplets = this.closeTriplets.bind(this);
+
+    helpers.addFluent(this.props.document);
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    const { message } = props;
+    if (message && message.id !== state.prevMessageId) {
+      const {
+        hasTriplets,
+        hasInterrupt,
+        interrupt,
+        triplets,
+        UTMTerm,
+      } = helpers.selectInterruptAndTriplets(message);
+
+      return {
+        prevMessageId: message.id,
+
+        hasInterrupt,
+        hasTriplets,
+
+        interrupt,
+        triplets,
+
+        isInterruptVisible: hasInterrupt,
+        isTripletsContainerVisible: hasTriplets,
+        isTripletsContentVisible: !(hasInterrupt || !hasTriplets),
```

Comment:
`hasTriplets && !hasInterrupt` seems like it might be easier to understand, but this is fine too